examples/red_rain.py

```diff
@@ -0,0 +1,181 @@
+        display.show(image)
+        sleep(600)
+        self._level += 1
+        self.time -= 5
+        self._misses = 0
+        self._hits = 0
+        display.scroll("L{} S{}".format(self._level, self.score))
+
+    def game_is_over(self):
+        if self._misses >= 5:
+            return True
+        return False
+
+
+class BallState(object):
+    # this tracks and controls the drops that fall from top of screen
+
+    _spawned = False
+    _direction = False
+    _column = 0
+    _row = 0
+    score = None
+
+    def _spawn(self):
+        if not self._spawned:
+            self._column = random.randint(0, 4)
+            self._row = -1
+            self._spawned = True
+            self._direction = False
+
+    def _move(self):
+        # falling ball in the field
+        if not self._direction and my_ball._row <= 3:
+            self._row += 1
+            return
+        # rising ball
+        if self._direction and my_ball._row > -1:
+            self._row -= 1
+            return
+
+    def _test_collision(self):
+        if self._row != 4:  # the ball is not low enough yet
+            return
+        if self._column == paddle_position or self._column == paddle_position + 1:  # player hit the ball
+            self._hit()
+        else:
+            self._miss()
+
+    def _despawn(self):
+        # get rid of a ball that went off the top or bottom of field
+        if not self._direction and my_ball._row == 4:
+            self._spawned = False
+        if self._direction and my_ball._row == -1:
+            self._spawned = False
+
+    def _hit(self):
+        # the player hit the ball, change its direction and register the hit
+        self._direction = True
+        self.score.hit()
+
+    def _miss(self):
+        # the player missed the ball, handle that miss and score it
+        self._despawn()
+        self.score.miss()
+
+    def run(self, score):
+        # this is the public entry
+        self._spawn()
+        self._move()
+        self._test_collision()
+        self._despawn()
+
+#    this method costs memory, only uncomment if needed
+#    def __str__(self):
+#        print('spa: ' + str(self._spawned) + ' row: ' + str(self._row) + ' column: ' + str(self._column) + ' dir: ' + str(self._direction))
+
+    def draw_field(self, row):
+        # draw one row of the field
+        if not self._spawned or not row is self._row:
+            return "00000:"
+        return ball_sprites[self._column]
+
+my_score = ScoreState()
+my_ball = BallState()
+my_ball.score = my_score  # skipping using __init__ since functions cost memory
+
+while True:
+    run_paddle()
+    my_ball.run(my_score)
+    field = [my_ball.draw_field(n) for n in range(4)]
+    field = ''.join(field) + paddle_sprites[paddle_position]
+    image = Image(field)
+    display.show(image)
+    my_score.level_complete()
+    if my_score.game_is_over():
+        break
+    sleep(my_score.time)
+
+display.scroll("G/O S {}".format(my_score.score), loop=True)
```

Comment:
G/O = Game over?

You can change the scroll speed with `delay=100` for example.